src/main/scala/vsys/blockchain/state/opcdiffs/TDBAOpcDiff.scala

```diff
@@ -131,12 +131,12 @@ object TDBAOpcDiff extends OpcDiffer {
         // relatedContract needed
         for {
           senderCallDiff <- getTriggerCallOpcDiff(context, OpcDiff.empty, sender, recipient, amount, tokenIdDataEntry, CallType.Trigger, 2)
-          senderRelatedAddress = if (sender.dataType == DataType.Address) Map(Address.fromBytes(sender.data).explicitGet() -> true) else Map[Address, Boolean]()
+          senderRelatedAddress = if (sender.dataType == DataType.Address) Map(Account.fromBytes(sender.data,0).explicitGet()._1 -> true) else Map[Account, Boolean]()
           senderDiff = OpcDiff(relatedAddress = senderRelatedAddress,
             tokenAccountBalance = Map(senderBalanceKey -> -transferAmount))
           senderTotalDiff = OpcDiff.opcDiffMonoid.combine(senderCallDiff, senderDiff)
           recipientCallDiff <- getTriggerCallOpcDiff(context, senderTotalDiff, sender, recipient, amount, tokenIdDataEntry, CallType.Trigger, 1)
-          recipientRelatedAddress = if (recipient.dataType == DataType.Address) Map(Address.fromBytes(recipient.data).explicitGet() -> true) else Map[Address, Boolean]()
```

Comment:
when u trying to deposit value to the contract account is also related to that contract account.

Reply:
yea, maybe delete the if condition here